cli/import_map.rs

```diff
@@ -0,0 +1,2133 @@
+use indexmap::IndexMap;
+use serde_json::Map;
+use serde_json::Value;
+use std::cmp::Ordering;
+use std::fs;
+use url::Url;
+
+#[derive(Debug)]
+pub struct ImportMapError {
+  pub msg: String,
+}
+
+impl ImportMapError {
+  pub fn new(msg: &str) -> Self {
+    ImportMapError {
+      msg: msg.to_string(),
+    }
+  }
+}
+
+// NOTE: here is difference between deno and reference implementation - deno currently
+//  can't resolve URL with other schemes (eg. data:, about:, blob:)
+const SUPPORTED_FETCH_SCHEMES: [&str; 3] = ["http", "https", "file"];
+
+type SpecifierMap = IndexMap<String, Vec<String>>;
+type ScopesMap = IndexMap<String, SpecifierMap>;
+
+#[derive(Debug)]
+pub struct ImportMap {
+  base_url: String,
+  imports: SpecifierMap,
+  scopes: ScopesMap,
+}
+
+impl ImportMap {
+  pub fn load(base_url: &str, file_name: &str) -> Result<Self, ImportMapError> {
+    let cwd = std::env::current_dir().unwrap();
+    let resolved_path = cwd.join(file_name);
+    debug!(
+      "Attempt to load import map: {}",
+      resolved_path.to_str().unwrap()
+    );
+
+    // Load the contents of import map
+    match fs::read_to_string(&resolved_path) {
+      Ok(json_string) => ImportMap::from_json(base_url, &json_string),
+      _ => panic!(
+        "Error retrieving import map file at \"{}\"",
+        resolved_path.to_str().unwrap()
+      ),
+    }
+  }
+
+  pub fn from_json(
```

Comment:
I wonder if it would be easier to use serde_derive to automatically generate a parser for this.
Might be a lot less code and reduce risk of bugs.

Reply:
I went that way, but after several passes I decided that it's actually easier and less code to do it manually.